Xamarin.Forms.Core/EnumerableExtensions.cs

```diff
@@ -25,22 +26,26 @@ public static class EnumerableExtensions
 				}
 		}
 
-		public static IEnumerable<T> GetGesturesFor<T>(this IEnumerable<IGestureRecognizer> gestures, Func<T, bool> predicate = null) where T : GestureRecognizer
+
+		public static IList<T> GetGesturesFor<T>(this IEnumerable<IGestureRecognizer> gestures, Func<T, bool> predicate = null) where T : GestureRecognizer
 		{
+			var result = new List<T>();
 			if (gestures == null)
-				yield break;
+				return result;
 
 			if (predicate == null)
 				predicate = x => true;
 
-			foreach (IGestureRecognizer item in gestures)
+			foreach (IGestureRecognizer item in new List<IGestureRecognizer>(gestures))
```

Comment:
this is what fixes the issue, right ? there's no need to change the return type to List<T>. Or am I missing something ?

Reply:
Like 103 alone fixes the issue. @hartez asked I change the return type to list. I obliged. Now, I'm suck! :)